File: Reqnroll/Chapter18/18.2-ResetBeforePattern/Before/WIMP.Specs/Support/Hooks.cs
using Reqnroll;

using WIMP.App.Data;
using WIMP.Specs.Drivers;

namespace WIMP.Specs.Support;

[Binding]
public class Hooks(AppHostingContext appHostingContext, IDatabaseDriver databaseDriver, IDataRepository dataRepository)
{
    [BeforeScenario(Order = 0)]
    public async Task CreateDatabase()
    {
        await databaseDriver.CreateDatabase();
    }

    [AfterScenario]
    public async Task DropDatabase()
    {
        await databaseDriver.DropDatabase();
    }

    [BeforeScenario(Order = 1)]
    public void CreateAppHost()
    {
        appHostingContext.AppHost = new WimpAppHost(dataRepository);
    }

    [AfterScenario]
    public void DisposeAppHost()
    {
        appHostingContext.AppHost?.Dispose();
    }
}

File: CucumberJVM/Chapter19/19.2-SupplementaryAttachmentPattern/Before/src/test/java/com/wimp/app/specs/support/Hooks.java
package com.wimp.app.specs.support;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Hooks {
    private static final Logger log = LoggerFactory.getLogger(Hooks.class);

    private final TestFileSystem testFileSystem;

    public Hooks(TestFileSystem testFileSystem) {
        this.testFileSystem = testFileSystem;
    }
}
